pilot/pkg/bootstrap/server.go

```diff
@@ -1347,6 +1352,29 @@ func (s *Server) serveHTTP() error {
 	return nil
 }
 
```

Comment:
One note - in testing I did still see errors but I think its from istiod shutting down too fast instead of becoming ready too soon. So kube-proxy was too slow.

Reply:
I've also noticed the same behavior. Without this fix, if adding the prestop hook, there won't be any downtime as well.